developer_tools/cargo/cargo.py

```diff
@@ -0,0 +1,501 @@
+        return process_not(re_dict["not_match"], rustc_cfg_dict)
+
+    if re_dict["basic_match"] is not None:
+        return process_basic(re_dict["basic_match"], rustc_cfg_dict)
+    return False
+
+
+def parse_target_format_platform(unparsed_platform, rustc_cfg_dict):
+    """
+    :param unparsed_platform: the string representation of the "target"-format
+    platform to parse
+    :type unparsed_platform: str
+    :param rustc_cfg_dict: dict containing information from the output of
+    `rustc --print cfg`
+    :type rustc_cfg_dict: dict
+    :returns: a bool indicating whether or not this "target"-format platform
+    should be included based on the contents of rustc_cfg_dict
+    :rtype: bool
+    """
+    if unparsed_platform == "---":
+        return True
+
+    target_components = unparsed_platform.split("-")
+    return (
+        len(target_components) == 4
+        and rustc_cfg_dict["target_arch"] == target_components[0]
+        and rustc_cfg_dict["target_vendor"] == target_components[1]
+        and rustc_cfg_dict["target_os"] == target_components[2]
+        and rustc_cfg_dict["target_env"] == target_components[3]
+    )
+
+
+def parse_platform(unparsed_platform):
+    """
+    :param unparsed_platform: the string representation of the platform to parse
+    :type unparsed_platform: str
+    :returns: a bool indicating whether or not this platform should be included
+    based on the contents of rustc_cfg_dict
+    :rtype: bool
+    """
+
+    cfg_pattern = r"cfg\(([^-]*)\)"
+    cfg_re = re.compile(cfg_pattern)
+    cfg_match = cfg_re.match(unparsed_platform)
+
+    rustc_cfg_dict = build_rustc_cfg_dict()
+
+    if cfg_match is not None:
+        return parse_cfg_format_platform(cfg_match, rustc_cfg_dict)
+
+    return parse_target_format_platform(unparsed_platform, rustc_cfg_dict)
+
+
+def build_cargo_outdated_dict():
+    """
+    :returns: a dictionary containing information from the output of `cargo
+    outdated`
+    the keys are the string representations of dependencies
+    the values are 4-tuples containing
+    1) the string represenation of the dependency's version (i.e. from the
+    "Project" column of the output of `cargo outdated`,
+    2) the string representation of the dependency the dependency is pulled in by
+    or None if the dependency is not pulled in by any dependency
+    3) the string representation of the dependency's platform information (i.e.
+    from the "Platform" column of the ouput of `cargo outdated`
+    4) a bool indicating whether or not the dependency should be "included", with
+    respect to the platform information
+    :rtype: dict
+    """
+    cargo_outdated_dict = {}
+
+    command = ["cargo", "outdated"]
+    proc = subprocess.Popen(command, stdout=subprocess.PIPE)
+
+    pattern = r"([^\s]*)\s*([^\s]*)\s*([^\s]*)\s*([^\s]*)\s*([^\s]*)\s*(.*)"
+    my_reg_ex = re.compile(pattern)
+
+    while True:
+        line_bo = proc.stdout.readline()
+
+        if not line_bo:
```

Comment:
Please avoid boolean overloading here and use an equality check instead. Does 'bo' stand for "boolean overloading"? Probably "line" is just as good as ```line_bo```.

Reply:
'bo' stands for byte object. `readline()` returns an object of the class 'bytes', and I later call `decode("utf-8")` in order to convert it into a string which I can perform regular expression on.

I'm not sure what you mean by "boolean overloading" or how I would use an equality check here; do you mind explaining further?

Renaming line_bo to line is reasonable.